pages/upload/upload.controller.js

```diff
@@ -47,28 +48,39 @@ function UploadController(api, session, $state, $translate) {
 
 		var data = {};
 
-		if (!self.loading && self.data && self.file) {
+		if (canUpload()) {
+			self.error = null;
 			data.title = self.data.title || undefined;
 			data.description = self.data.description || undefined;
-			data.tag = [];
+			data.productType = [];
 			
 			if (self.data.productType) {
 				for (var type in self.data.productType) {
 					if(self.data.productType[type]) {
-						data.tag.push(type);
+						data.productType.push(type);
 					}
 				}
 			}
 
-			data.tag = data.tag.toString();
+			data.productType = data.productType.toString();
 
 			self.loading = true;
 			api.upload(api.url + '/video', self.file, data, function() {
 				self.loading = false;
 				$state.go('gallery');
 			});
 		} else {
-			self.error = 'No video selected or data is not set';
+			self.error = $translate.instant('UPLOAD_ERROR_DATA');
 		}
 	}
+
+	function canUpload() {
```

Comment:
great for the fix! :+1: 

not sure if product types were to be mandatory in the end, please check it with @PabloFMM